base/dates/parse.jl

```diff
@@ -281,9 +281,14 @@ function Base.tryparse{T<:TimeType}(
     pos, len = start(str), endof(str)
     values, pos = tryparsenext_internal(T, str, pos, len, df, false)
     if isnull(values)
-        Nullable{T}()
+        return Nullable{T}()
     else
-        Nullable{T}(T(unsafe_get(values)...))
+        try
+            parsed = Nullable{T}(T(unsafe_get(values)...))
```

Comment:
I wonder whether there's an API we could use for now where we have `isvalid` return both a flag and an error string if called with an optional `Val{true}` or something, as a way of avoiding the code duplication without needing try-catches.